Content fixes 2024-12

scr0llbaer9000 opened the pull request:
The small corrections as recently discussed.

Find some more mistakes, validator can generally flag `{` without the `$` in front, it will remain unlikely that we'll have / need curly brackets `{ }` in text.

Corrected some long-standing anachronisms.

Added a couple of new items that were in the oven.

nittka commented on res/database/Default/database_news.xml:
```diff
@@ -6904,22 +6982,45 @@ W tym roku ponownie spodziewane są demonstracje.</pl>
 			</description>
 			<variables>
-				<borisbecker>${.person:"209467cd-85e1-4212-999e-b6273359320f":"fullname"}</borisbecker>
+				<athletesfirstname_csv>${.person:"614c90b8-e18c-45d2-a026-a3d216b276cb":"firstname"};${.person:"c6f7756d-3e0c-4e47-9cbe-13dc7826e145":"firstname"};${.person:"b073ffa5-ad2f-40ce-ba52-b5b9d5e4a2a6":"firstname"};${.person:"209467cd-85e1-4212-999e-b6273359320f":"firstname"}</athletesfirstname_csv>
```
You could just put the GUID in the csv and use a single expression for first- and last name. This would increase readability and possibly consistency as both GUID and function will not be duplicated.

scr0llbaer9000 replied:
i do this sometimes, i'm still experimenting with different approaches in such cases.

nittka commented on res/database/Default/user/scr0llbaer_news.xml:
```diff
@@ -2260,5 +2260,1642 @@
+				<en>Italy introduce "National Siesta Day" to boost productivity. Critics call it "lazy Friday". Businesses complain about rising pillow costs.</en>
+				<pl>Włochy wprowadziły „Narodowy Dzień Sjesty“ dla zwiększenia produktywności. Krytycy nazywają go „leniwy piątek“. Przedsiębiorcy narzekają na rosnące koszty poduszek.</pl>
+			</description>
+			<data genre="0" price="1" quality_min="5" quality_max="40" fictional="1" />
```
unique? Without variables this item (and following) might be repeated often.
Another option would be effects ensuring they are not available for a couple of years.

scr0llbaer9000 replied:
it shares the thread id with similar items, which will be my approach in such cases.

nittka replied:
OK.

GWRon commented on res/database/Default/database_ads.xml:
```diff
@@ -148,7 +148,20 @@ Naprawdę trzeba żuć.</pl>
+			</description>
+			<conditions min_audience="0.39" min_image="4" target_group="2" pro_pressure_groups="0" contra_pressure_groups="0" />
+			<data quality="10" repetitions="2" duration="3" profit="2000" penalty="3220" fix_infomercial_profit="1" />
+			<availability year_range_to="2000" />
```
Availability ranges are "exclusive"
from=2000
to=2001

means: 
any year _smaller_ than 2000 is excluded
any year _bigger_ than 2001 is excluded
-> 2000 _and_ 2001 are allowed.

In your case this means:
Sticker Fuzzy is available from X till (including) 1999.
Ringtone Fuzzy is available from 2002 till X.

scr0llbaer9000 replied:
are you sure? it's wrong everywhere then.
even before "my time" it was done that way (e.g. "PoliPlay")

nittka replied:
Note that the return value for isAvailable is False in the snippet. So actually, the years defined in the database are inclusive. (If currentYear >2000 Then not available.)

GWRon commented on res/database/Default/user/jorgaeff.xml:
```diff
@@ -428,21 +430,19 @@
-				<de>Aufgrund des erstmaligen Abstiegs des ${sportsclub_hamburgersv} zieht Präsident Uwe Gehler die Reißleine und entlässt Trainer Knut Knolle. Geeler möchte nun den Trainermarkt sondieren und nach geeigneten Nachfolgern Ausschau halten.</de>
-				<en>Due to the first-time relegation of ${sportsclub_hamburgersv}, president Uwe Gehler pulls the plug and sacks coach Knut Knolle. Geeler now wants to explore the market and look for suitable successors.</en>
-				<pl>Z powodu pierwszego spadku ${sportsclub_hamburgersv}, prezes Uwe Gehler wyciąga wtyczkę i zwalnia trenera Knuta Knolle. Geeler chce teraz zbadać rynek i poszukać odpowiednich następców.</pl>
+				<de>Aufgrund des erstmaligen Abstiegs des ${sportsclub_hamburgersv} zieht Präsident ${uweseeler} die Reißleine und entlässt Trainer Knut Knolle. Geeler möchte nun den Trainermarkt sondieren und nach geeigneten Nachfolgern Ausschau halten.</de>
```
"Geeler möchte nun den Trainermarkt" - hier braucht es noch eine Referenz

scr0llbaer9000 commented:
Applied fixes and enhancements.

scr0llbaer9000 commented:
hope I caught them all now

GWRon commented on res/database/Default/user/sjaele.xml:
```diff
@@ -348,21 +348,6 @@
 			<availability year_range_from="" year_range_to="" />
 		</news>
 
-		<news guid="Sjaele-news-Plagiatorin-Forschungsausschuss" type="0" creator="7430" created_by="Sjaele">
```
Where did the news go to?

scr0llbaer9000 replied:
to `database_news.xml` where real historical news are now listed chronologically.

GWRon replied:
Odd. Onl viewed it via github/browser and explicitly opened/uncollapsed the common news xml too but ctrl+f did not find it except in the file where it was removed.

Sorry for the fuzz

scr0llbaer9000 commented:
Betty rating is now applied to "Kultur Heute" (but have to do another test run if it behaves like desired).

Added a few more screenplay-scripts by eggi and ronny.

Added a few more news items. Some sports ones; can be used for referencing sports players eventually.

scr0llbaer9000 commented on res/database/Default/user/eggi_screenplays_adaptations.xml:
```diff
@@ -79,5 +79,272 @@
+			<data flags="4" target_groups="96" />
+			<targetgroupattractivity men="0.75" teenagers_male="0.25" unemployed="0.25" />
+			<genres maingenre="6" subgenres="11,300" />
+			<episodes min="11" max="11" />
```
ok removing here, though I still have that attribute for many "static" screenplay-scripts. Not removing there for now.

scr0llbaer9000 commented on res/database/Default/database_scripts.xml:
```diff
@@ -1975,6 +1976,419 @@ Ekscytująca rozrywka i łamiące mózg pytania quizowe.</pl>
+				</scripttemplate>
+				<scripttemplate index="27" product="7" licence_type="2" guid="e9e67487-33f7-4ef9-ae5a-ec49e6e1966f"> <!-- 'dupe' but other explanation -->
+					<title>
+						<de>Warum fliegen Flugzeuge?</de>
```
yeah but different description. Noticed that too and commented. All in all repetitions cannot be ruled out for the more dynamic templates either. So leaving as is.

GWRon replied:
The issue is, that a "random selection" can randomly (heh...) select both "variants" for one series. Read: you might end up with a series of 5 episodes in which 2 are "duplicates" (variants).

For now there are only two options to avoid things:
- possible now: create 2 scrips with each of them only containing 1 variant (but of course this reduces "variety" if you also split up the ones only existing in one variation)
- requiring efforts: flags/markers telling that an episode cannot "co-exist" with others. You would eg. mark the variants as to exclude each other. This can be useful if you eg want to offer multiple "variants" of a series pilot.

Ok there is another: removing the duplicates (and saving them for later)

scr0llbaer9000 commented:
Final corrections, that's all for now. Not intending to add more things. Can be merged from my side.

scr0llbaer9000 commented on res/database/Default/user/cujo.xml:
```diff
@@ -322,7 +322,7 @@
 		<!-- "Stiftung Warentest" -->
 		<!--TODO variables-->
-		<news guid="cujo-news-stiftung-guetertest-01" type="0" thread_guid="cujo-news-stiftung-guetertest" creator="8827" created_by="Cujo">
+		<news guid="cujo-news-stiftung-guetertest-01" type="0" creator="8827" created_by="Cujo">
```
I think something is not working as intended with these. It seems the other news item won't show up, and the other will show up only once.

But I thought we can examine that after the release.

nittka replied:
OK

scr0llbaer9000 commented on res/database/Default/database_programmes_fictional.xml:
```diff
@@ -223,16 +223,16 @@
 			<groups target_groups="0" pro_pressure_groups="0" contra_pressure_groups="0" />
-			<targetgroupattractivity men="0.5" />
-			<data country="D" distribution="2" maingenre="103" subgenre="300,6" flags="5" blocks="1" broadcast_flags="512" price_mod="10" />
+			<targetgroupattractivity men="0.25" women="0.25" />
```
Yes thought that too, got 20% and then 10%. 10% would be okay-ish.
